src/include/kdbtypes.h

```diff
@@ -85,7 +89,7 @@ typedef unsigned char             kdb_octet_t;
 typedef signed short              kdb_short_t;
 typedef unsigned short            kdb_unsigned_short_t;
 
-#if SIZEOF_LONG == 4
+#if (LONG_MAX == +2147483647)
```

Comment:
as mentioned before, it works on my OS X setup:

	//Result of TRY_COMPILE
	HAVE_SIZEOF_LONG:INTERNAL=TRUE
	//Result of TRY_COMPILE
	HAVE_SIZEOF_LONG_DOUBLE:INTERNAL=TRUE
	//Result of TRY_COMPILE
	HAVE_SIZEOF_LONG_LONG:INTERNAL=TRUE

the values are set as expected on a 64 bit machine

	//CHECK_TYPE_SIZE: sizeof(long)
	SIZEOF_LONG:INTERNAL=8
	//CHECK_TYPE_SIZE: sizeof(long long)
	SIZEOF_LONG_LONG:INTERNAL=8